builtin/providers/aws/resource_aws_opsworks_rds_db_instance.go

```diff
@@ -0,0 +1,183 @@
+package aws
+
+import (
+	"log"
+	"time"
+
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/opsworks"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsOpsworksRdsDbInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsOpsworksRdsDbInstanceRegister,
+		Update: resourceAwsOpsworksRdsDbInstanceUpdate,
+		Delete: resourceAwsOpsworksRdsDbInstanceDeregister,
+		Read:   resourceAwsOpsworksRdsDbInstanceRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"stack_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"rds_db_instance_arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"db_password": &schema.Schema{
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"db_user": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceAwsOpsworksRdsDbInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).opsworksconn
+
+	d.Partial(true)
+
+	// @TODO if these two params force new resources, is the following necessary?
+	if d.HasChange("rds_db_instance_arn") || d.HasChange("stack_id") {
+		return fmt.Errorf("cannot change 'rds_db_instance_arn' and 'stack_id' for opsworks rds db. changes on these attributes will force a new resource")
+	}
+
+	d.SetPartial("rds_db_instance_arn")
+	req := &opsworks.UpdateRdsDbInstanceInput{
+		RdsDbInstanceArn: aws.String(d.Get("rds_db_instance_arn").(string)),
+	}
+
+	requestUpdate := false
+	if d.HasChange("db_user") {
+		d.SetPartial("db_user")
+		req.DbUser = aws.String(d.Get("db_user").(string))
+		requestUpdate = true
+	}
+	if d.HasChange("db_password") {
+		d.SetPartial("db_password")
+		req.DbPassword = aws.String(d.Get("db_password").(string))
+		requestUpdate = true
+	}
+
+	if true == requestUpdate {
+		log.Printf("[DEBUG] Opsworks RDS DB Instance Modification request: %s", req)
```

Comment:
Is there any need to wait here to make sure that the instance modifications have been completed? Otherwise we could get into a situation that terraform is trying to make the same changes again